commons/statistics/ols.go

```diff
@@ -0,0 +1,28 @@
+package statistics
+
+type Point interface {
+	X() float64
+	Y() float64
+}
+
+func LeastSquares(series []Point) (m float64, b float64) {
+	n := float64(len(series))
+
+	if n == 0 {
+		return
+	}
+
+	var sumx, sumy, sumxx, sumxy float64
+
+	for _, p := range series {
+		x, y := p.X(), p.Y()
+		sumx += x
```

Comment:
For numerical accuracy, we should subtract either the mean or the first value from the X and Y samples, then add those back in to _b_ before returning.